tests/cache/TEST_PLAN_CACHE_SET.md

```diff
@@ -0,0 +1,274 @@
+# Test Plan for CacheSet
```

Comment:
You call it test plan and I got really confused for a sec lol, but these are just unit tests